ts/annotations/MockPoint.ts

```diff
@@ -0,0 +1,541 @@
+/* *
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+import H from '../parts/Globals.js';
+
+/**
+ * Internal types.
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        interface AnnotationMockLabelOptionsObject {
+            x: (number|null);
+            y: (number|null);
+            point: AnnotationMockPoint;
+        }
+        class AnnotationMockPoint {
+            public static fromPoint(point: AnnotationPoint): AnnotationMockPoint;
+            public static pointToOptions(point: AnnotationPointType): AnnotationMockPointOptionsObject;
+            public static pointToPixels(point: AnnotationPointType, paneCoordinates?: boolean): PositionObject;
+            public constructor(
+                chart: AnnotationChart,
+                target: (AnnotationControllable|null),
+                options: (AnnotationMockPointOptionsObject|Function)
+            );
+            public command?: string;
+            public isInside: boolean;
+            public mock: true;
+            public options: (AnnotationMockPointOptionsObject|Function);
+            public plotX: number;
+            public plotY: number;
+            public series: AnnotationMockSeries;
+            public target: (AnnotationControllable|null);
+            public visible?: boolean;
+            public x: (number|null);
+            public y: (number|null);
+            public applyOptions(options: AnnotationMockPointOptionsObject): void;
+            public getLabelConfig(): AnnotationMockLabelOptionsObject;
+            public getOptions(): AnnotationMockPointOptionsObject;
+            public hasDynamicOptions(): boolean;
+            public isInsidePane(): boolean;
+            public refresh(): void;
+            public refreshOptions(): void;
+            public rotate(cx: number, cy: number, radians: number): void;
+            public scale(cx: number, cy: number, sx: number, sy: number): void;
+            public setAxis(options: AnnotationMockPointOptionsObject, xOrY: ('x'|'y')): void;
+            public toAnchor(): Array<number>;
+            public translate(cx: (number|undefined), cy: (number|undefined), dx: number, dy: number): void
+        }
+        interface AnnotationMockPointOptionsObject {
+            command?: string;
+            series?: undefined;
+        }
+        interface AnnotationMockSeries {
+            chart: AnnotationChart;
+            getPlotBox: Series['getPlotBox'];
+            xAxis?: (Axis|null);
+            yAxis?: (Axis|null);
+            visible: boolean;
+        }
+        interface AnnotationPoint {
+            mock?: undefined;
```

Comment:
Actually `AnnotationPoint` is just `Highcharts.Point`. The way to distinguish between `Point` and `MockPoint` is that `MockPoint.mock` is defined and set to true. Should we then replace `AnnotationPoint` with `Point`?

Reply:
See previous comment.